gui/src-tauri/src/main.rs

```diff
@@ -92,10 +92,11 @@ fn main() {
 
 	// Spawn server process
 	let exit_flag = Arc::new(AtomicBool::new(false));
-	let mut backend: Option<CommandChild> = None;
+	let backend = Arc::new(Mutex::new(Option::<CommandChild>::None));
```

Comment:
I'm not very happy with this, but its out of the scope of this PR. I think I will submit a PR to refactor this, its becoming messier and messier and this code should all be dead simple.

Reply:
thats honestly understandable, its a hack rn